syntax = "proto3";

package tasks;

service TaskService {
    rpc GetTasks (Empty) returns (TaskList);
    rpc CreateTask (Task) returns (Task);
}

message Task {
    int32 id = 1;
    string title = 2;
    string description = 3;
    string priority = 4;
    string status = 5;
    string created_at = 6;
    string updated_at = 7;
}

message TaskList {
    repeated Task tasks = 1;
}

message Empty {}